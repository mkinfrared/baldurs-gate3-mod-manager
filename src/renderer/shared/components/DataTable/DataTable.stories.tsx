import { Meta, StoryObj } from "@storybook/react";

import DataTable from "./DataTable";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/DataTable",
  component: DataTable,
  tags: ["autodocs"],
} satisfies Meta<typeof DataTable>;

const Default: Story = {
  args: {
    data: [],
    columns: [],
  },
  render: (args) => (
    <div className={css.Story}>
      <DataTable {...args} />
    </div>
  ),
};

export { Default };

export default meta;
