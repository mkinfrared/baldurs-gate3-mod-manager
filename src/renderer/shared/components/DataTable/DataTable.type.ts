import { ColumnDef } from "@tanstack/react-table";

export type DataTableProps<T> = {
  columns: ColumnDef<T, any>[];
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  data: Array<T>;
};
