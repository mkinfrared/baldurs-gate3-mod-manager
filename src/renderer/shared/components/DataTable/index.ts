export { default } from "./DataTable";

export * from "./DataTable.type";
