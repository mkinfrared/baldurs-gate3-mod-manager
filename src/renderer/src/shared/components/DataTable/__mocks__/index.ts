export { default } from "./DataTable.mock";
