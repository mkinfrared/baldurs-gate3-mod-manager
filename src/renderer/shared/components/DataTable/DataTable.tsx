import {
  flexRender,
  getCoreRowModel,
  useReactTable,
} from "@tanstack/react-table";

import { classNames } from "renderer/shared/lib/helpers";
import { Table } from "renderer/shared/ui";

import css from "./DataTable.module.scss";
import { DataTableProps } from "./DataTable.type";

const DataTable = <T,>({ className, columns, data }: DataTableProps<T>) => {
  const table = useReactTable({
    columns,
    data,
    getCoreRowModel: getCoreRowModel(),
  });

  return (
    <Table
      className={classNames(css.DataTable, className)}
      data-testid="DataTable"
    >
      <thead>
        {table.getHeaderGroups().map((headerGroup) => (
          <tr key={headerGroup.id}>
            {headerGroup.headers.map((header) => (
              <th key={header.id}>
                {header.isPlaceholder
                  ? null
                  : flexRender(
                      header.column.columnDef.header,
                      header.getContext(),
                    )}
              </th>
            ))}
          </tr>
        ))}
      </thead>
      <tbody>
        {table.getRowModel().rows.map((row) => (
          <tr key={row.id}>
            {row.getVisibleCells().map((cell) => (
              <td key={cell.id}>
                {flexRender(cell.column.columnDef.cell, cell.getContext())}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default DataTable;
