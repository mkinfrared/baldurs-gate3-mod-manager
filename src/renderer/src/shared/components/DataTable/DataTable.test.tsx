import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import DataTable from "./DataTable";

describe("<DataTable />", () => {
  const data: any[] = [];
  const columns: any[] = [];
  const Component = <DataTable data={data} columns={columns} />;

  it("should be defined", () => {
    expect(DataTable).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
