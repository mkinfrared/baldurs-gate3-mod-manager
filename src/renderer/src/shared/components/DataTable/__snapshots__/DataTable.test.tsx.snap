// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DataTable /> > should match the snapshot 1`] = `
<div>
  <table
    class="Table DataTable"
    data-testid="Table"
  >
    <thead>
      <tr />
    </thead>
    <tbody />
  </table>
</div>
`;
